pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -352,7 +352,12 @@ func makeFirewallName(name string) string {
 // TODO(a-robinson): Don't just ignore specified IP addresses. Check if they're
 // owned by the project and available to be used, and use them if they are.
 func (gce *GCECloud) EnsureTCPLoadBalancer(name, region string, externalIP net.IP, ports []*api.ServicePort, hosts []string, affinityType api.ServiceAffinity) (*api.LoadBalancerStatus, error) {
-	glog.V(2).Info("Checking if load balancer already exists: %s", name)
+	if len(hosts) == 0 {
+		glog.V(2).Error("Call to EnsureTCPLoadBalancer with no hosts")
+		return nil, errors.New("Cannot EnsureTCPLoadBalancer() with no hosts")
```

Comment:
Responding to @lavalamp's comment: "It seems like a problem if the old load balancer points to a bunch of things that no longer exist?"

The node sync loop part of the service controller already handles resizing the target pools down as the nodes in the cluster change. Not deleting the load balancer entirely enables a use case where you can scale down your cluster's nodes to zero but still have the same IP address for the service if you later re-add any nodes.